File: test_api.js
const http = require('http');

const loginOptions = {
  hostname: 'localhost',
  port: 8080,
  path: '/api/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded'
  }
};

const req = http.request(loginOptions, (res) => {
  let cookie = res.headers['set-cookie'];
  
  const listOptions = {
    hostname: 'localhost',
    port: 8080,
    path: '/api/tickets/list',
    method: 'GET',
    headers: {
      'Cookie': cookie ? cookie.join(';') : ''
    }
  };

  const listReq = http.request(listOptions, (listRes) => {
    let data = '';
    listRes.on('data', (chunk) => data += chunk);
    listRes.on('end', () => {
      console.log('STATUS:', listRes.statusCode);
      console.log('BODY:', data);
    });
  });
  listReq.end();
});
req.write('username=Anmol0001&password=Anmol0001');
req.end();

File: test_api.py
import requests

session = requests.Session()
# Get CSRF token
r = session.get('http://localhost:8080/')
csrf_token = session.cookies.get('XSRF-TOKEN')

# Login
login_data = {'username': 'Anmol0001', 'password': 'Anmol0001'}
headers = {'X-XSRF-TOKEN': csrf_token}
r_login = session.post('http://localhost:8080/api/auth/login', data=login_data, headers=headers)
print("Login Status:", r_login.status_code)

# Get tickets
r_tickets = session.get('http://localhost:8080/api/tickets/list')
print("Tickets Status:", r_tickets.status_code)
print("Tickets Body:", r_tickets.text)
